tests/pds/dataservice_test.go

```diff
@@ -627,6 +628,88 @@ var _ = Describe("{UpgradeDataServiceImage}", func() {
 	})
 })
 
+var _ = Describe("{CordonNodeAndDeletePod}", func() {
+	JustBeforeEach(func() {
+		StartTorpedoTest("CordonNodeAndDeletePod", "Deploys a data service, cordons one selected node, deletes the pod from that node", pdsLabels, 0)
+	})
+
+	It("Cordon a Node and Delete Pods from that node", func() {
+		Step("Deploy, Validate, Cordon Node and Delete Pods, Validate new Pod, Validate Storage, Run Workload on Data Service", func() {
+			for _, ds := range params.DataServiceToTest {
+				// Run this Test on Consul only. We can enable more Data Services later from here if needed.
+				// Consul is a random choice. We can choose any other Data service as well.
+				// This test case is pretty much Data service independent
+				if ds.Name != consul {
+					log.Warnf("Skipping %v dataservice ", ds.Name)
+					continue
+				}
+				isDeploymentsDeleted = false
+				deployment, _, _, err = DeployandValidateDataServices(ds, tenantID, projectID)
+				log.FailOnError(err, "Error while deploying data services")
+				log.InfoD("Running Check Node DataService Func %v ", ds.Name)
+				nodes, err := pdslib.GetNodesOfSS(*deployment.ClusterResourceName, namespace)
+				log.FailOnError(err, "Cannot fetch nodes of the running Data Service")
+				for _, node := range nodes {
+					log.InfoD("Node Name is: %s", node.Name)
+				}
+				var nodeName string
+				Step("Delete Pods and Cordon a node", func() {
+					nodeName = nodes[0].Name // Selecting the 1st node in the list to cordon
+					podsList, err := pdslib.GetPodsOfSsByNode(*deployment.ClusterResourceName, nodeName, namespace)
+					log.FailOnError(err, "Pod not found on this Node")
+					log.InfoD("Pods found on %v node. Trying to Cordon and Drain pods from this node now.", nodeName)
+					err = k8sCore.DrainPodsFromNode(nodeName, podsList, timeOut, maxtimeInterval)
+					log.FailOnError(err, "draining pod from the node failed")
+				})
+				Step("Validate Data Service Again", func() {
+					resourceTemp, storageOp, config, err := pdslib.ValidateDataServiceVolumes(deployment, ds.Name, dataServiceDefaultResourceTemplateID, storageTemplateID, namespace)
+					log.FailOnError(err, "error on ValidateDataServiceVolumes method")
+					ValidateDeployments(resourceTemp, storageOp, config, ds.Replicas, dataServiceVersionBuildMap)
+				})
+				Step("Validate no pods are on the cordoned node anymore", func() {
+					nodes, err := pdslib.GetNodesOfSS(*deployment.ClusterResourceName, namespace)
+					log.FailOnError(err, "Cannot fetch nodes of the running Data Service")
+					for _, nodeObj := range nodes {
+						log.InfoD("Node Name is: %s", nodeObj.Name)
+						if nodeObj.Name == nodeName {
```

Comment:
do u need this data to be pushed to dashboard?

Reply:
Thanks - removed this log line